lib/GIL/Function.cpp

```diff
@@ -0,0 +1,48 @@
+#include "Function.hpp"
+
+namespace glu::gil {
+
+BasicBlock *Function::popFirstBlock()
+{
+    if (_basicBlocks.empty())
+        return nullptr;
+
+    BasicBlock *bb = &_basicBlocks.front();
+    _basicBlocks.remove(bb);
+
+    return bb;
+}
```

Comment:
this seems useless. we also never want a basic block to exist outside of a function, so returning a basic block that was removed from a function is dangerous.